File: 1. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ans;
        unordered_map<int,int> table;
        for(int i=0;i<nums.size();i++){
            if(table[target-nums[i]]){
                ans.push_back(table[target-nums[i]]-1);
                ans.push_back(i);
            }
            table[nums[i]]=i+1;
        }
        return ans;
    }
};

File: 1. Two Sum.go
func twoSum(nums []int, target int) []int {
    var ans []int
    table := make(map[int]int)
    for i:=0;i<len(nums);i++{
        if table[target-nums[i]]>0{
            ans=append(ans,table[target-nums[i]]-1)
            ans=append(ans,i)
        }else{
            table[nums[i]]=i+1;
        }
    }
    return ans
}